.github/actions/backend-deploy-job/files/cloudbuild.yaml

```diff
@@ -1,188 +1,251 @@
+            gcloud scheduler jobs create http ${job_name}-scheduler-trigger-$((++schedule_id)) \
+              --project="${job_project}" \
+              --location "${_REGION}" \
+              --time-zone="PST" \
+              --schedule="${line}" \
+              --uri="https://${_REGION}-run.googleapis.com/apis/run.googleapis.com/v1/namespaces/${job_project}/jobs/${job_name}:run" \
+              --http-method POST \
+              --oauth-service-account-email sa-job@${job_project}.iam.gserviceaccount.com
+          fi
+        done
+      fi
+    }
 
-      for file in run_*sh;
-      do
-          JOB_SCRIPT_NAME=$(echo "${file##*/}")
-          export JOB_NAME=$(echo ${_APP_NAME}-${_DEPLOY_TARGET}-${JOB_SCRIPT_NAME} | cut  -d'.' -f 1 | tr "_" "-")
-          export COMMAND="./${JOB_SCRIPT_NAME}"
+    if [[ -n "$(ls run_*.sh 2> /dev/null)" ]]; then
+      gcloud run jobs describe "${_APP_NAME}-${_DEPLOYMENT_ENV}" --format=yaml --project="${PROJECT_ID}" --region="${_REGION}" > ./devops/gcp/k8s/job.${_DEPLOYMENT_ENV}.yaml
+      yq -i 'del(.status, .metadata.generation, .metadata.creationTimestamp, .metadata.namespace, .metadata.resourceVersion, .metadata.selfLink, .metadata.uid)' ./devops/gcp/k8s/job.${_DEPLOYMENT_ENV}.yaml
 
-          # replace job and
-          yq e '.metadata.name = env(JOB_NAME) | .spec.template.spec.template.spec.containers[0].command = env(COMMAND)' \
-            ./devops/gcp/k8s/job.${_DEPLOY_TARGET}.yaml > ./devops/gcp/k8s/${JOB_NAME}.yaml
+      for file in run_*sh; do
+        JOB_SCRIPT_NAME="${file##*/}"
+        echo "Job ${JOB_SCRIPT_NAME}"
 
-          # create job if not exist
-          res=$(gcloud run jobs describe $JOB_NAME --project=${JOB_NAMESPACE} --region=${_REGION} || echo "NOT_EXIST")
+        export JOB_NAME=$(echo ${_APP_NAME}-${_DEPLOYMENT_ENV}-${JOB_SCRIPT_NAME} | cut -d'.' -f 1 | tr "_" "-")
+        export COMMAND="./${JOB_SCRIPT_NAME}"
 
-          if [ "$res" = "NOT_EXIST" ]; then
-            echo "Creating your job..."
-            gcloud run jobs create $JOB_NAME --project=${JOB_NAMESPACE} --region=${_REGION}
-          fi
+        yq e '.metadata.name = env(JOB_NAME) | .spec.template.spec.template.spec.containers[0].command = env(COMMAND)' \
+          ./devops/gcp/k8s/job.${_DEPLOYMENT_ENV}.yaml > ./devops/gcp/k8s/${JOB_NAME}.yaml
 
-          # replace the yaml
-          gcloud run jobs replace ./devops/gcp/k8s/${JOB_NAME}.yaml --project=${JOB_NAMESPACE} --region=${_REGION}
+        if ! gcloud run jobs describe "${JOB_NAME}" --project="${PROJECT_ID}" --region="${_REGION}" &>/dev/null; then
+          echo "Creating your job..."
+          IMAGE="${_REGION}-docker.pkg.dev/${_BUILD_PROJECT}/cloud-run-repo/${_APP_NAME}:${_DEPLOYMENT_ENV}"
+          gcloud run jobs create "${JOB_NAME}" --project="${PROJECT_ID}" --region="${_REGION}" --image="${IMAGE}"
+        fi
 
+        gcloud run jobs replace ./devops/gcp/k8s/${JOB_NAME}.yaml --project="${PROJECT_ID}" --region="${_REGION}"
```

Comment:
Possible we could add in adhoc jobs? I had another PR for that, sometimes OPS needs to run the jobs with edited container settings

Reply:
adhoc jobs should be use different way to do it. https://github.com/bcgov/bcregistry-gcp-jobs